# Projet 2 : Analysez des données de systèmes éducatifs #

### Import du nécessaire ###

In [10]:
import numpy as np
import pandas as pd
from annexe import annexe as an
import re
pd.set_option('display.max_rows', 50000)
pd.set_option('display.max_columns', 50000)
pd.set_option('display.width', 1000000)
pd.set_option('display.max_colwidth', None)

## 1. Ouverture et affichage grossier du contenu des fichiers du dataset ##

In [9]:

for num_fichier in range(len(an.FICHIERS)):
    print("Fichier numéro ",num_fichier + 1," :")
    data_frame = an.ouvre_csv(num_fichier)
    
    an.print_samples(data_frame,10)
    print("\n\n\n\n\n")

Fichier numéro  1  :
EdStatsCountry.csv  is open
fichier avec  241  lignes et  31  colonnes.



Country Code                                    Short Name                  Table Name                                     Long Name 2-alpha code     Currency Unit                                                                                                                                                                                                                           Special Notes                     Region          Income Group WB-2 code                         National accounts base year  National accounts reference year                SNA price valuation Lending category Other groups                                     System of National Accounts Alternative conversion factor PPP survey year             Balance of Payments Manual in use External debt Reporting status       System of trade    Government Accounting concept             IMF data dissemination standard Latest population census                            Latest household survey           Source of most recent Income and expenditure data Vital registration complete Latest agricultural census  Latest industrial data  Latest trade data Latest water withdrawal data
117          LAO                                       Lao PDR                     Lao PDR              Lao People's Democratic Republic           LA           Lao kip                                                                                                                                                                                                                                     NaN        East Asia & Pacific   Lower middle income        LA                                                2002                               NaN  Value added at basic prices (VAB)              IDA          NaN  Country uses the 1993 System of National Accounts methodology.                           NaN            2005  IMF Balance of Payments Manual, 6th edition.                    Preliminary  Special trade system     Budgetary central government                                         NaN                     2005  Multiple Indicator Cluster Survey (MICS), 2011/12              Expenditure survey/budget survey (ES/BS), 2008                         NaN                    2010/11                     NaN                NaN                         2005
8            ARM                                       Armenia                     Armenia                           Republic of Armenia           AM     Armenian dram                                                                                                                                                                                                                                     NaN      Europe & Central Asia   Lower middle income        AM  Original chained constant price data are rescaled.                            1996.0  Value added at basic prices (VAB)             IBRD          NaN  Country uses the 1993 System of National Accounts methodology.                       1990–95            2005  IMF Balance of Payments Manual, 6th edition.                         Actual  Special trade system  Consolidated central government  Special Data Dissemination Standard (SDDS)                     2011          Demographic and Health Survey (DHS), 2010                     Integrated household survey (IHS), 2012                         Yes                    2013/14                     NaN             2012.0                         2007
148          MNA  Middle East & North Africa (developing only)  Middle East & North Africa  Middle East & North Africa (developing only)           XQ               NaN                                                                                                                                               Middle East and North Africa regional aggregate (does not include high-income economies).                        NaN                   NaN        XQ                                                 NaN                 







Fichier numéro  2  :
EdStatsCountry-Series.csv  is open
fichier avec  613  lignes et  3  colonnes.



CountryCode         SeriesCode                                                                                                                                                                  DESCRIPTION
129         CYP     NY.GDP.PCAP.KD                                                                                  Data are for areas under the effective control of the Government of the Republic of Cyprus.
130         CZE        SP.POP.TOTL                                                                                                                                                      Data sources : Eurostat
146         EGY        SP.POP.TOTL                                                                                                                     Data sources : United Nations World Population Prospects
328         MDA        SP.POP.TOTL  Data sources : National Bureau of Statistics. Excluding Transnistria. For 1950-94, World Bank estimates using UN World Population Prospects' growth rates of whole Moldova.
6           AFG  NY.GNP.MKTP.PP.CD                                                                                                                                           Estimates are based on regression.
414         PAN        SP.POP.TOTL                                                                                                                     Data sources : United Nations World Population Prospects
293         LBY  NY.GDP.PCAP.PP.KD                                                                                                                                           Estimates are based on regression.
456         SAU        SP.POP.TOTL                                                                                                                     Data sources : United Nations World Population Prospects
572         URY        SP.POP.GROW                                                                                                                      Data sources: United Nations World Population Prospects
565         TZA     NY.GNP.PCAP.CD                                                                                                                                               Covers mainland Tanzania only.







Fichier numéro  3  :
EdStatsData.csv  is open
fichier avec  886930  lignes et  69  colonnes.



Country Name Country Code                                                                                                                 Indicator Name           Indicator Code  1970  1971  1972  1973  1974  1975  1976  1977  1978  1979  1980  1981  1982  1983  1984  1985  1986  1987  1988  1989  1990  1991  1992  1993  1994  1995  1996  1997  1998      1999      2000      2001      2002           2003  2004  2005           2006      2007          2008           2009       2010        2011          2012      2013      2014      2015  2016  2017   2020   2025   2030   2035   2040  2045  2050  2055  2060  2065  2070  2075  2080  2085  2090  2095  2100
580192                 Morocco          MAR                                                                        Enrolment in early childhood education, female (number)                UIS.E.0.F   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  276729.0  284978.0  272226.0  269767.0  265455.000000   NaN   NaN  283008.000000  288762.0  278243.00000  305610.000000  284019.00         NaN  290284.00000  293823.0  326036.0       NaN   NaN   NaN    NaN    NaN    NaN    NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
860979                 Vanuatu          VUT                                                          Wittgenstein Projection: Mean Years of Schooling. Age 25+. Gender Gap         PRJ.MYS.25UP.GPI   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN       NaN       NaN       NaN       NaN            NaN   NaN   NaN            NaN       NaN           NaN            NaN       0.88         NaN           NaN       NaN       NaN    0.7600   NaN   NaN   0.65   0.55   0.45   0.37   0.31  0.26  0.22  0.20  0.18  0.17  0.16  0.15  0.15  0.15  0.14  0.14  0.14
456219              Kazakhstan          KAZ                                                        PASEC: Distribution of 2nd grade language scores: 99th Percentile Score       LO.PASEC.REA.2.P99   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN       NaN       NaN       NaN       NaN            NaN   NaN   NaN            NaN       NaN           NaN            NaN        NaN         NaN           NaN       NaN       NaN       NaN   NaN   NaN    NaN    NaN    NaN    NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
727430                Slovenia          SVN                                                     PASEC: Distribution of 2nd grade mathematics scores: 10th Percentile Score       LO.PASEC.MAT.2.P10   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN       NaN       NaN       NaN       NaN            NaN   NaN   NaN            NaN       NaN           NaN            NaN        NaN         NaN           NaN       NaN       NaN       NaN   NaN   NaN    NaN    NaN    NaN    NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
336527                 Finland          FIN                                                          TIMSS: Distribution of 8th Grade Science Scores: 5th Percentile Score        LO.TIMSS.SCI8.P05   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN     407.0       NaN       NaN       NaN            NaN   NaN   NaN            NaN       NaN           NaN            NaN        NaN  444.000000           NaN       NaN       NaN       NaN   NaN   NaN    NaN    NaN    NaN    NaN    NaN   NaN   NaN   NaN   NaN   Na







Fichier numéro  4  :
EdStatsFootNote.csv  is open
fichier avec  643638  lignes et  4  colonnes.



CountryCode            SeriesCode    Year                                     DESCRIPTION
296894         KIR           SH.DYN.MORT  YR2017          5-year average value between 0s and 5s
642793         ZWE       UIS.OE.56.40510  YR2002  UNESCO Institute for Statistics (UIS) estimate
31126          AZE        IT.NET.USER.P2  YR2008                              Population age 7+.
183055         EST     SE.ENR.TERT.FM.ZS  YR1996                                    Country Data
277370         JAM  SE.SEC.ENRL.VO.FE.ZS  YR2002                             Country estimation.
306492         KWT        SE.ADT.LITR.ZS  YR1985                                    Country data
15842          ARE     SE.PRM.REPT.FE.ZS  YR1995                                    Country Data
560262         SWE           SE.PRM.ENRL  YR2002                                       Estimated
548804         SSF        UIS.NERT.2.GPI  YR2006  UNESCO Institute for Statistics (UIS) estimate
501966         QAT           SE.PRM.ENRL  YR1998                                    Country Data







Fichier numéro  5  :
EdStatsSeries.csv  is open
fichier avec  3665  lignes et  15  colonnes.



Series Code               Topic                                                                                                           Indicator Name                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Short definition                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Long definition Periodicity                 Base Period Other notes Aggregation method Limitations and exceptions General comments                                                                                                Source Statistical concept and methodology Development relevance Related source links
1818        PRJ.ATT.4064.3.FE          Attainment   Projection: Percentage of the population age 40-64 by highest level of educational attainment. Upper Secondary. Female                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            NaN                                                                                                                Share of the population of the stated age group that has completed upper secondary or incomplete post-secondary education as the highest level of educational attainment. Projections are based on collected census and survey data for the base year (around 2010) and the Medium Shared Socioeconomic Pathways (SSP2) projection model. The SSP2 is a middle-of-the-road scenario that combines medium fertility with medium mortality, medium migration, and the Global Education Trend (GET) education scenario. For more information and other projection models, consult the Wittgenstein Centre for Demography and Global Human Capital's website: http://www.oeaw.ac.at/vid/dataexplorer/         NaN  Projections (2010 to 2100)         NaN                NaN                        NaN              NaN  Wittgenstein Centre for Demography and Global Human Capital: http://www.oeaw.ac.at/vid/dataexplorer/                                 NaN                   NaN                  NaN
2883     UIS.EA.5T8.AG25T99.M          Attainment    UIS: Percentage of population age 25+ with at least a completed short-cycle tertiary degree (ISCED 5 or higher). Male                                                                                                  

#### Rapide analyse des 5 fichiers
**1. EdstatCountry.csv**  
On peut voir qu'il y a dans le premier fichier **31 colonnes de 241 lignes**.  
Dans les colonnes nous trouvons:
 - des identifiants : Country Code,Short Name,Table Name,Long Name,2-alpha code,WB-2 code
 - des categories : Lending category, Currency Unit, Income Group, Government Accounting concept, IMF data dissemination standard
 - des données géographiques : Region
 - des rescencements : National accounts base year, National accounts reference year, SNA price valuation, PPP survey year, Latest population census, Latest household survey, Latest agricultural census, Latest industrial data, Latest trade data, Latest water withdrawal data
 - d'autres données : Special Notes, Other groups,System of National Accounts,Alternative conversion factor,Balance of Payments Manual in use,External debt Reporting status, System of trade, Source of most recent Income and expenditure data, Vital registration complete 

Ce fichier semble être une source de données générals sur chaque pays.
 
 
**2. EdStatsCountry-Series.csv**  
On peut voir qu'il y a dans le deuxième fichier **3 colonnes de 613 lignes**.  
Dans les colonnes nous trouvons:
 - des identifiants : CountryCode, SeriesCode.
 - une description : DESCRIPTION.

Ce fichier donne la source des données pour une série à un pays donné.


**3. EdStatsData.csv**  
On peut voir qu'il y a dans le troisième fichier EdStatsData.csv **69 colonnes de 886930 lignes**.  
Dans les colonnes nous trouvons:
 - des identifiants : Country Name, Country Code, Indicator Name, Indicator Code.
 - des années : de 1970 à 2017, puis de 2020 à 2100 par pas de 5 ans.

Ce fichier donne les nombres pour chaque année de chaque série.


**4. EdStatsFootNote.csv**  
On peut voir qu'il y a dans le quatrième fichier **4 colonnes de 643638 lignes**.  
Dans les colonnes nous trouvons:
 - des identifiants : CountryCode, SeriesCode.
 - une donnée temporelle : Year.
 - une description : DESCRIPTION.

Ce fichier donne la provenance des données quantitative pour une série, un pays à une année donné.


**5. EdStatsSeries.csv**
On peut voir qu'il y a dans le cinquième et dernier fichier **15 colonnes de 3665 lignes**.  
Dans les colonnes nous trouvons:
 - des identifiants : Series Code.
 - des descriptions : Indicator Name, Topic, Short definition, Long definition.
 - des données temporelles : Periodicity, Base Period.
 - d'autres données : Other notes, Aggregation method, Limitations and exceptions, General comments, Source, Statistical concept and methodology, Development relevance, Related source links.

Ce fichier semble être une source de données descriptives de chaque série .


Après avoir regardé tout les fichiers il ne me parait pas utile de continuer l'exploration dans les fichiers EdStatsCountry-Series.csv et EdStatsFootNote.csv. Le fichier EdstatCountry.csv me parait utile pour la colonne 'Income group' de chaque pays, je vais donc le garder pour plus tard.
Je vais donc continuer avec les fichiers EdStatsData.csv pour les données chiffrées. Il me semble que EdStatsSeries.csv contient des explications sur les lignes de EdStatsData.csv, si besoin j'utiliserais donc EdStatsSeries.csv pour mieux comprendre EdStatsData.csv.

Je vois dans EdStatsData.csv que l'on parle d'éducation, de primaire, secondaire, peut-être tertiaire. On parle aussi d'années moyennes de scolarité.

## 2. Recherche de ligne contenant des mots clé

Je vais essayer de faire ressortir tout les différents type de lignes (colonne:'Indicator Name') du fichier EdStatsData.csv contenant : school, student, secondary, tertiary, educational, inhabitant, household, technology, computer, Internet.

In [7]:
fichierESD = an.ouvre_csv(2)
index = fichierESD.loc[:,'Indicator Name'].squeeze()
index_secondary = index.str.contains('econdary', regex = True, na = False)
index_tertiary = index.str.contains('ertiary', regex = True, na = False)
index_school_educational = index.str.contains('chool|ducationnal', regex = True, na = False)#('school&primary{0}&secondary{0}&tertiary{0}|educational&primary{0}&secondary{0}&tertiary{0}', regex = True, na = False)
index_student = index.str.contains('student', regex = True, na = False)
index_inhabitant_household= index.str.contains('nhabitant|ousehold', regex = True, na = False)
index_tech_computer_internet= index.str.contains('echnology|omputer|nternet', regex = True, na = False)
# print(np.random.choice(fichierESD[reindex]['Indicator Name'].unique(),1000))
print(len(fichierESD[index_secondary]['Indicator Name'].unique()))
print(len(fichierESD[index_tertiary]['Indicator Name'].unique()))
print(len(fichierESD[index_school_educational]['Indicator Name'].unique()))
print(len(fichierESD[index_student]['Indicator Name'].unique()))
print(len(fichierESD[index_inhabitant_household]['Indicator Name'].unique()))
print(len(fichierESD[index_tech_computer_internet]['Indicator Name'].unique()))

EdStatsData.csv  is open
fichier avec  886930  lignes et  69  colonnes.

847
330
675
684
52
34


Les mots school et educationnal ne sont pas présent sans secondary ou tertiary, je peux donc enlever ces mots clé de ma recherche.
847
330
600
684
52
26

### 2.1 Regard des résultats

Je vais regarder pour chaque mot clé les lignes obtenues et prendre celles qui nous interessent sans trop faire de tri. Je ne mettrais pas deux fois les lignes qui intersecte deux catégories.

In [87]:
print(fichierESD[index_secondary]['Indicator Name'].unique())

['Adjusted net enrolment rate, lower secondary, both sexes (%)'
 'Adjusted net enrolment rate, lower secondary, female (%)'
 'Adjusted net enrolment rate, lower secondary, gender parity index (GPI)'
 'Adjusted net enrolment rate, lower secondary, male (%)'
 'Adjusted net enrolment rate, upper secondary, both sexes (%)'
 'Adjusted net enrolment rate, upper secondary, female (%)'
 'Adjusted net enrolment rate, upper secondary, gender parity index (GPI)'
 'Adjusted net enrolment rate, upper secondary, male (%)'
 'Africa Dataset: Graduates from accredited pre-service lower secondary teacher training programmes, both sexes (number)'
 'Africa Dataset: Graduates from accredited pre-service lower secondary teacher training programmes, female (number)'
 'Africa Dataset: Graduates from accredited pre-service primary or secondary teacher training programmes, both sexes (number)'
 'Africa Dataset: Graduates from accredited pre-service primary or secondary teacher training programmes, female (numbe

Les catégories intéressantes sont :  
-Adjusted net enrolment rate, upper secondary, both sexes (%)  
-Barro-Lee: Average years of secondary schooling, age 15+, total  
-Barro-Lee: Average years of secondary schooling, age 25+, total  
-Barro-Lee: Percentage of population age 15+ with secondary schooling. Completed Secondary  
-Current expenditure as % of total expenditure in secondary public institutions (%)  
-Enrolment in Grade 6 of lower secondary general education, both sexes (number)  
-Enrolment in post-secondary non-tertiary education, both sexes (number)  
-Enrolment in secondary education, both sexes (number)  
-Gross enrolment ratio, upper secondary, both sexes (%)  
-School life expectancy, secondary, both sexes (years)  

In [88]:
print(fichierESD[index_tertiary]['Indicator Name'].unique())

['All staff compensation as % of total expenditure in post-secondary non-tertiary public institutions (%)'
 'All staff compensation as % of total expenditure in tertiary public institutions (%)'
 'Barro-Lee: Average years of tertiary schooling, age 15+, female'
 'Barro-Lee: Average years of tertiary schooling, age 15+, total'
 'Barro-Lee: Average years of tertiary schooling, age 15-19, female'
 'Barro-Lee: Average years of tertiary schooling, age 15-19, total'
 'Barro-Lee: Average years of tertiary schooling, age 20-24, female'
 'Barro-Lee: Average years of tertiary schooling, age 20-24, total'
 'Barro-Lee: Average years of tertiary schooling, age 25+, female'
 'Barro-Lee: Average years of tertiary schooling, age 25+, total'
 'Barro-Lee: Average years of tertiary schooling, age 25-29, female'
 'Barro-Lee: Average years of tertiary schooling, age 25-29, total'
 'Barro-Lee: Average years of tertiary schooling, age 30-34, female'
 'Barro-Lee: Average years of tertiary schooling, age 30-34

Les catégories intéressantes sont :  
-Adjusted net enrolment rate, upper secondary, both sexes (%)  
-Barro-Lee: Average years of tertiary schooling, age 25+, total  
-Barro-Lee: Percentage of population age 15+ with tertiary schooling. Completed Tertiary  
-Current expenditure as % of total expenditure in tertiary public institutions (%)  
-Enrolment in tertiary education per 100,000 inhabitants, both sexes  
-Enrolment in tertiary education, ISCED 5 programmes, both sexes (number)  
-Enrolment in tertiary education, ISCED 6 programmes, both sexes (number)  
-Enrolment in tertiary education, ISCED 7 programmes, both sexes (number)  
-Enrolment in tertiary education, ISCED 8 programmes, both sexes (number)  
-Graduates from tertiary education, both sexes (number)  
-UIS: Percentage of population age 25+ with a completed short-cycle tertiary degree (ISCED 5). Total  
-Gross enrolment ratio, tertiary, both sexes (%)  
-School life expectancy, tertiary, both sexes (years)  

In [83]:
print(fichierESD[index_school_educational]['Indicator Name'].unique())

['Africa Dataset: Average number of grades per multigrade class in primary schools (number of grades)'
 'Africa Dataset: Average size of classes in primary schools (number of pupils)'
 'Africa Dataset: Average size of multigrade classes in primary schools (number of pupils)'
 'Africa Dataset: Average size of single grade classes in Grade 1 of primary schools (number of pupils)'
 'Africa Dataset: Average size of single grade classes in Grade 2 of primary schools (number of pupils)'
 'Africa Dataset: Average size of single grade classes in Grade 3 of primary schools (number of pupils)'
 'Africa Dataset: Average size of single grade classes in Grade 4 of primary schools (number of pupils)'
 'Africa Dataset: Average size of single grade classes in Grade 5 of primary schools (number of pupils)'
 'Africa Dataset: Average size of single grade classes in Grade 6 of primary schools (number of pupils)'
 'Africa Dataset: Average size of single grade classes in Grade 7 of primary schools (number o

Les catégories intéressantes sont :  
-UIS: Percentage of population age 25+ with no schooling. Total  
-Wittgenstein Projection: Mean years of schooling. Age 25+. Total  
-UIS: Mean years of schooling of the population age 25+. Total  
-Wittgenstein Projection: Percentage of the population age 25+ by highest level of educational attainment. Post Secondary. Total  
-Wittgenstein Projection: Percentage of the population age 25+ by highest level of educational attainment. Upper Secondary. Total  
-Wittgenstein Projection: Percentage of the total population by highest level of educational attainment. Upper Secondary. Total  
-Wittgenstein Projection: Population age 20-24 in thousands by highest level of educational attainment. Upper Secondary. Total  
-Wittgenstein Projection: Population age 25-29 in thousands by highest level of educational attainment. Post Secondary. Total  
-Wittgenstein Projection: Population in thousands by highest level of educational attainment. Upper Secondary. Total  
-Wittgenstein Projection: Population in thousands by highest level of educational attainment. Post Secondary. Total  

In [89]:
print(fichierESD[index_student]['Indicator Name'].unique())

['EGRA: Identification of the Initial Sound of a Spoken Word - Share of students with a zero score (%). Bamanankan. 2nd Grade'
 'EGRA: Identification of the Initial Sound of a Spoken Word - Share of students with a zero score (%). Bomu. 2nd Grade'
 'EGRA: Identification of the Initial Sound of a Spoken Word - Share of students with a zero score (%). Chichewa. 2nd Grade'
 'EGRA: Identification of the Initial Sound of a Spoken Word - Share of students with a zero score (%). Chichewa. 4th Grade'
 'EGRA: Identification of the Initial Sound of a Spoken Word - Share of students with a zero score (%). English. 2nd Grade'
 'EGRA: Identification of the Initial Sound of a Spoken Word - Share of students with a zero score (%). English. 3rd Grade'
 'EGRA: Identification of the Initial Sound of a Spoken Word - Share of students with a zero score (%). English. 4th Grade'
 'EGRA: Identification of the Initial Sound of a Spoken Word - Share of students with a zero score (%). English. 6th Grade'
 'EGRA

Il n'y a aucune nouvelle cellule interessante dans ce jeu de données.

In [85]:
print(fichierESD[index_inhabitant_household]['Indicator Name'].unique())

['Enrolment in tertiary education per 100,000 inhabitants, both sexes'
 'Enrolment in tertiary education per 100,000 inhabitants, female'
 'Enrolment in tertiary education per 100,000 inhabitants, male'
 'SABER: (School Finance) Policy Goal 2 Lever 2: Do payments for schooling represent a high share of income for low income households?'
 'UIS: Rate of out-of-school adolescents of lower secondary school age, both sexes (household survey data) (%)'
 'UIS: Rate of out-of-school adolescents of lower secondary school age, female (household survey data) (%)'
 'UIS: Rate of out-of-school adolescents of lower secondary school age, fourth quintile, both sexes (household survey data) (%)'
 'UIS: Rate of out-of-school adolescents of lower secondary school age, fourth quintile, female (household survey data) (%)'
 'UIS: Rate of out-of-school adolescents of lower secondary school age, fourth quintile, male (household survey data) (%)'
 'UIS: Rate of out-of-school adolescents of lower secondary scho

Il n'y a aucune nouvelle cellule interessante dans ce jeu de données.

In [6]:
index_population= index.str.contains('population', regex = True, na = False)
# print(np.random.choice(fichierESD[reindex]['Indicator Name'].unique(),1000))
print(len(fichierESD[index_population]['Indicator Name'].unique()))
print(fichierESD[index_population]['Indicator Name'].unique())

509
['Adult illiterate population, 15+ years, % female'
 'Adult illiterate population, 15+ years, both sexes (number)'
 'Adult illiterate population, 15+ years, female (number)'
 'Adult illiterate population, 15+ years, male (number)'
 'Adult literacy rate, population 15+ years, both sexes (%)'
 'Adult literacy rate, population 15+ years, female (%)'
 'Adult literacy rate, population 15+ years, gender parity index (GPI)'
 'Adult literacy rate, population 15+ years, male (%)'
 'Barro-Lee: Percentage of female population age 15+ with no education'
 'Barro-Lee: Percentage of female population age 15+ with primary schooling. Completed Primary'
 'Barro-Lee: Percentage of female population age 15+ with primary schooling. Total (Incomplete and Completed Primary)'
 'Barro-Lee: Percentage of female population age 15+ with secondary schooling. Completed Secondary'
 'Barro-Lee: Percentage of female population age 15+ with secondary schooling. Total (Incomplete and Completed Secondary)'
 'Barro-Le

Les catégories intéressantes sont :  
-UIS: Percentage of population age 25+ with a completed short-cycle tertiary degree (ISCED 5). Total  
-UIS: Percentage of population age 25+ with a completed bachelor's or equivalent degree (ISCED 6). Total  
-UIS: Percentage of population age 25+ with a completed master's or equivalent degree (ISCED 7). Total  
-UIS: Percentage of population age 25+ with at least a completed short-cycle tertiary degree (ISCED 5 or higher). Total  
-UIS: Percentage of population age 25+ with at least a completed bachelor's or equivalent degree (ISCED 6 or higher). Total  
-UIS: Percentage of population age 25+ with at least a completed master's degree or equivalent (ISCED 7 or higher). Total  
-UIS: Percentage of population age 25+ with a doctoral degree or equivalent (ISCED 8). Total  
-UIS: Percentage of population age 25+ with at least completed post-secondary education (ISCED 4 or higher). Total  


In [8]:
print(fichierESD[index_tech_computer_internet]['Indicator Name'].unique())

['Internet users (per 100 people)'
 'Percentage of graduates from non-science and non-technology fields in tertiary education who are female (%)'
 'Percentage of graduates from Science and Technology programmes in tertiary education who are female (%)'
 'Personal computers (per 100 people)'
 'PIAAC: Adults by proficiency level in problem solving in technology-rich environments (%). Below Level 1'
 'PIAAC: Adults by proficiency level in problem solving in technology-rich environments (%). Failed the ICT Core Test'
 'PIAAC: Adults by proficiency level in problem solving in technology-rich environments (%). Level 1'
 'PIAAC: Adults by proficiency level in problem solving in technology-rich environments (%). Level 2'
 'PIAAC: Adults by proficiency level in problem solving in technology-rich environments (%). Level 3'
 'PIAAC: Adults by proficiency level in problem solving in technology-rich environments (%). No computer experience'
 'PIAAC: Adults by proficiency level in problem solving in

Les catégories intéressantes sont : 
-Internet users (per 100 people)  
-Personal computers (per 100 people)

### 2.2 Tri des données intéressentes

Je vais faire un second tri pour déterminer une stratégie avec les catégories ressorties au point d'avant.  
Plusieurs catégories peuvent être créé en sommant les données entre elles.  
  
Nombres de personnes entrant en études dans les catégories qui nous interessent:  
-Enrolment in Grade 6 of lower secondary general education, both sexes (number)  
-Enrolment in post-secondary non-tertiary education, both sexes (number)  
-Enrolment in secondary education, both sexes (number)  
-Enrolment in tertiary education, ISCED 5 programmes, both sexes (number)  
-Enrolment in tertiary education, ISCED 6 programmes, both sexes (number)  
-Enrolment in tertiary education, ISCED 7 programmes, both sexes (number)  
-Enrolment in tertiary education, ISCED 8 programmes, both sexes (number) 
  
Nombre d'années d'études moyenne par personne agé de plus de 25 ans:  
-Barro-Lee: Average years of secondary schooling, age 25+, total  
-Barro-Lee: Average years of tertiary schooling, age 25+, total  
  
-UIS: Mean years of schooling of the population age 25+. Total 
  
Nombre d'années d'études que voudrait faire les étudiants:  
-School life expectancy, secondary, both sexes (years)  
-School life expectancy, tertiary, both sexes (years) 
  
  
Nombre d'étudiants ayant fini des études de troisième cycle:   
-Graduates from tertiary education, both sexes (number)  

Prévision en millier du nombre d'étudiants:  
-Wittgenstein Projection: Population age 20-24 in thousands by highest level of educational attainment. Upper Secondary. Total  
-Wittgenstein Projection: Population age 25-29 in thousands by highest level of educational attainment. Post Secondary. Total  



Nombre d'utilisateur d'internet pour 100 personnes:  
-Internet users (per 100 people)  

Nombre d'ordinateurs personnel pour 100 personnes:  
-Personal computers (per 100 people)  

Les catégories enlevé étaient soient redondante dans l'information qu'elles apportaient soit ne donnaient que des fragments que ce que d'autre catégories apportent.

## 3.Préparation des dataframes

### 3.1 Enlever les colonnes inintéressantes 

Je veux garder dans le fichier **EdStatsCountry** 'Country Code','Short Name' et 'Income Group'.  
Pour le fichier **EdStatsData** je veux garder les colonnes 'Country Name','Country Code','Indicator Name' et les colonnes des années entre 2001 et 2031.

In [12]:
ESC_column_to_keep = ['Country Code','Short Name','Income Group']
ESD_column_to_keep = ['Country Name','Country Code','Indicator Name'] + [range(2001,2017,1)] + [range(2020,2030,5)]

fichierESC = an.ouvre_csv(0,column_list=ESC_column_to_keep)
fichierESD = an.ouvre_csv(2,column_list=ESD_column_to_keep)

TypeError: ouvre_csv() got an unexpected keyword argument 'column_list'

# 2. Création d'un dataframe exclusivement chiffré ***************** ancienne version

Un dataframe exclusivement chiffré va me permettre grâce à la librairie pandas de faire des opérations plus efficace. Je vais donc prendre le fichier qui nous intéresse ( EdStatsData.csv), mettre en index une colonne à identifiant unique (Indicator Code sera testé) et enlever les colonnes contenant des données autre que numériques.

In [92]:
fichier3 = an.ouvre_csv(2)
an.print_info_col(fichier3,["Indicator Code"])

EdStatsData.csv  is open
fichier avec  886930  lignes et  69  colonnes.


Indicator Code
3665 n uniques
0 somme des naN dans la colonne


La calonne Indicator Code n'est pas unique, je vais donc prendre comme identifiant la colonne Indicator Code concaténé à la colonne Country Name.

fichier3 = an.ouvre_csv(2 ,nom_colonne = ["Indicator Code","Country Name"])

In [93]:
# fichier3 = an.ouvre_csv(2)
# fichier3 = an.ouvre_csv(2,nom_colonne = "Indicator Code")
fichier3['id'] = fichier3['Indicator Code'] + '_' + fichier3['Country Name']
fichier3 = fichier3.set_index('id')
fichier3JustNb = fichier3.drop(columns = ["Country Code","Indicator Name","Indicator Code","Country Name"])

In [94]:
fichier3JustNb.sample(11)

1970  1971  1972  1973  1974  1975  1976  1977  1978  1979  1980  1981  1982  1983  1984  1985  1986  1987  1988  1989   1990  1991  1992  1993  1994   1995  1996  1997  1998  1999   2000  2001  2002  2003  2004  2005  2006  2007  2008       2009   2010  2011      2012  2013  2014       2015  2016  2017  2020  2025  2030  2035  2040  2045  2050  2055  2060  2065  2070  2075  2080  2085  2090  2095  2100
id                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
LO.EGRA.CLSPM.CIN.2GRD_Singapore                                    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN        NaN    NaN   NaN       NaN   NaN   NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
HH.DHS.SCR.Q3_Qatar                                                 NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN        NaN    NaN   NaN       NaN   NaN   NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
LO.EGRA.NCWPM.DAN.2GRD_France                                       NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN        NaN    NaN   NaN       NaN   NaN   NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
LO.PASEC.REA.6.P1_West Bank and Gaza                                NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN        NaN    NaN   NaN       NaN   NaN   NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
HH.DHS.YRS.15UP.GIN.Q3_Maldives                                     NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  45.695212    NaN   NaN       NaN   NaN   NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
LO.EGRA.ORF.ENG.6GRD_Least developed countries: UN classification   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN        NaN    NaN   NaN       NaN   NaN   NaN        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN
UIS.OAPP.1_Low income                                               NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   NaN    NaN   NaN   NaN   NaN   

Notre dataframe a maintenant un identifiant unique (les colonnes Country Name et Indicator Code) et ne contient que des nombres ou des valeurs non assigné.

## 3. Epuration pas à pas du dataframe 

On va enlever les colonnes qui paraissent trop loin dans le temps que se soit dans le passé ou dans le futur. Puis on explorera le fichier pour voir si il y a des lignes vide et dans ce cas les enlever. On va chercher si il y a des differents ordre de grandeur de valeur dans le dataframe. On pourra aussi rechercher des pourcentage pour plus tard afficher des statistiques.

### 1. Enlever les colonnes inutile pour l'analyse

In [4]:
range(1970,2001,1) 
fichier3JustNb = fichier3JustNb.drop(['1970','1971','1972','1973','1974','1975','1976','1977','1978','1979','1980','1981','1982','1983','1984','1985','1986','1987','1988','1989','1990','1991','1992','1993','1994','1995','1996','1997','1998','1999','2000','2035','2040','2045','2050','2055','2060','2065','2070','2075','2080','2085','2090','2095','2100'], axis = 1)
fichier3JustNb.shape

(886930, 20)

Il ne reste plus que 20 colonnes sur les 65 dates du début.

### 2. Enlever les lignes vides

Je vais voir si j'arrive a afficher des lignes vides en faisant la somme cumulé des valeur naN pour chaque ligne.

In [8]:
tabIsnaCumsum = fichier3JustNb.isna().cumsum(axis=1)
an.print_samples(tabIsnaCumsum,10)

                                                    2001  2002  2003  2004  \
id                                                                           
SABER.SCH.FNNC.GOAL4.LVL2_Seychelles                   1     2     3     4   
LO.TIMSS.MAT4.MA_Least developed countries: UN ...     1     2     3     4   
HH.MICS.OOST.X.M_Middle East & North Africa            1     2     3     4   
HH.DHS.YRS.1519.Q5_Europe & Central Asia (exclu...     1     2     3     4   
UIS.E.2.PU.F_Burundi                                   1     2     2     2   
LO.EGRA.LSTN.0.ENG.2GRD_Denmark                        1     2     3     4   
SE.SEC.ENRL.FE_Virgin Islands (U.S.)                   1     2     3     4   
PRJ.ATT.2064.1.MA_Greenland                            1     2     3     4   
LO.PASEC.REA.2.FE_Barbados                             1     2     3     4   
SABER.TECH.GOAL6_Austria                               1     2     3     4   

                                                    2005  2006 

Nous avons 20 colonnes, donc quand on voit le nombre 20 dans la dernière colonnes (2030) c'est que la ligne est vide, on peut donc la supprimer.

In [6]:
fichier3JustNb2 = fichier3JustNb.dropna(axis = 'index',how = 'all')
fichier3JustNb2.shape

(347465, 20)

Sur nos 886930 lignes du départ, il ne reste plus que 347465 lignes, ce qui nous a bien allégé le dataframe.

In [75]:
# tabIsnaCumsum2 = fichier3JustNb2.isna().cumsum(axis=1)
# an.print_samples(tabIsnaCumsum2,10)

## 3. Exploration des ordres de grandeur du fichier

Je vais afficher le min,le max et l'écart type que je trouve dans chaque colonnes.

In [56]:
print("min")
print(fichier3JustNb2.min())
print("max")
print(fichier3JustNb2.max())
print("ecart-type")
print(fichier3JustNb2.std())

min
2001    -69737.000000
2002    -92143.000000
2003   -111186.000000
2004   -126260.000000
2005   -134724.000000
2006   -370894.000000
2007   -388217.000000
2008   -408854.000000
2009   -456124.000000
2010   -496905.000000
2011   -570994.000000
2012   -604993.000000
2013   -615748.000000
2014       -89.000000
2015        -2.467847
2016        -1.797539
2017         1.000000
2020        -1.870000
2025        -2.000000
2030        -2.110000
dtype: float64
max
2001    6.477834e+13
2002    6.654753e+13
2003    6.906328e+13
2004    7.278829e+13
2005    7.624045e+13
2006    8.031840e+13
2007    8.469145e+13
2008    8.711022e+13
2009    8.677516e+13
2010    9.134676e+13
2011    9.506314e+13
2012    9.999473e+13
2013    1.054580e+14
2014    1.108060e+14
2015    1.156198e+14
2016    1.206030e+14
2017    4.000000e+00
2020    1.599479e+06
2025    1.781898e+06
2030    1.967258e+06
dtype: float64
ecart-type
2001    5.449242e+11
2002    5.622886e+11
2003    5.803607e+11
2004    6.278337e+11
2005   

Les max sont énorme, les min sont extrèmement négatif et l'écart type très grand. Il y a donc des données qui n'ont pas le même ordre de grandeur et ne sont pas comparables.  
Je vais donc faire des catégories de lignes en fonction de l'ordre de grandeur. Je vais pour l'instant essayé de sortir des pourcentage ceux compris entre 1 et 100, ceux entre 0 et 1. Je vais mettre de coté les valeurs négatives et les valeurs supérieur à 100.

In [7]:
fichier3pow2plus = an.part_dataframe(fichier3JustNb2,100)
fichier3LessThanPow2 = an.part_dataframe(fichier3JustNb2,1,100)
fichier3lessThanOne = an.part_dataframe(fichier3JustNb2,0,1)
fichier3lessThanZero = an.part_dataframe(fichier3JustNb2,0,-1)
# fichier3JustNb2.iloc[fichier3JustNb2[:] > 10000].samples(10)
# fichier3pow10plus = fichier3JustNb2[fichier3JustNb2.ge(10000000000)].dropna(axis = 'index',how = 'all')
# fichier3pow7plus = fichier3JustNb2[fichier3JustNb2.lt(10000000000) & fichier3JustNb2.gt(10000000)].dropna(axis = 'index',how = 'all')
# fichier3pow2plus = fichier3JustNb2[fichier3JustNb2.gt(100)].dropna(axis = 'index',how = 'all')
# fichier3LessThanPow2 = fichier3JustNb2[fichier3JustNb2.lt(100) & fichier3JustNb2.gt(1)].dropna(axis = 'index',how = 'all')
# fichier3lessThanOne = fichier3JustNb2[fichier3JustNb2.lt(1) & fichier3JustNb2.gt(0)].dropna(axis = 'index',how = 'all')
# fichier3lessThanZero = fichier3JustNb2[fichier3JustNb2.lt(0)].dropna(axis = 'index',how = 'all')
# print(fichier3JustNb2[fichier3JustNb2.ge(10000000000) == True] )
# fichier3pow10plus = fichier3JustNb2[fichier3JustNb2[:] > 999999999 ]
# print(fichier3pow10plus.shape)
# print(fichier3pow7plus.shape)
# fichier3pow2plus2 = fichier3JustNb2.join(fichier3pow2plus,how ='left',lsuffix=['2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009','2010', '2011', '2012', '2013', '2014', '2015', '2016', '2017', '2020','2025', '2030'])
# print(fichier3pow2plus2.shape)
# print(fichier3pow2plus2.shape)
print(fichier3pow2plus.shape)
print(fichier3LessThanPow2.shape)
print(fichier3lessThanOne.shape)
print(fichier3lessThanZero.shape)


(101796, 40)
(192098, 40)
(69746, 40)
(582, 40)


In [43]:
        
def find_def(df_to_define,df_with_def):
    columns_name = ["Indicator Code","Country Name"]
    list_of_index = df_to_define.index.values
    newDataFrame = pd.DataFrame(columns = columns_name)
    for i,index in enumerate(list_of_index):
        index_couple = index.split("_")
        newDataFrame.loc[i] = index_couple
#     print(newDataFrame.shape)
#     print(newDataFrame.sample(10))
    newDataFrame.merge(df_with_def,left_on = 'Indicator Code',right_on = 'Series Code')   
        
EdStatDataSeries = EdStatData.merge(EdStatSeries,left_on = 'Indicator Code',right_on = 'Series Code')
#     for elem in list_of_index:
#         list_of_id_and_country = elem.split(separator = '_')
#         print(list_of_id_and_country)

find_def(fichier3lessThanZero)
        

(582, 2)
            Indicator Code     Country Name
194           UIS.MENFR.56         Djibouti
203       PRJ.MYS.25UP.GPI          Ecuador
338            SP.POP.GROW        Lithuania
508       PRJ.MYS.15UP.GPI        St. Lucia
314       PRJ.MYS.15UP.GPI  Kyrgyz Republic
30             UIS.MENF.56       Azerbaijan
431       PRJ.MYS.15UP.GPI           Panama
104  LO.PASEC.REA.6.RU.GAP          Burundi
166            UIS.MENF.56       Costa Rica
453       PRJ.MYS.25UP.GPI      Puerto Rico


In [11]:
print(fichier3pow2plus[fichier3pow2plus.lt(100)].dropna(axis = 'index',how = 'all'))

                                  2001['2001', '2002', '2003', '2004', '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016', '2017', '2020', '2025', '2030']  \
id                                                                                                                                                                                                       
SE.PRM.ENRR.MA_Arab World                                                 95.380608                                                                                                                      
SE.PRM.GINT.ZS_Arab World                                                 91.057167                                                                                                                      
SE.PRM.GINT.MA.ZS_Arab World                                              94.116013                                                                                                             

Affichons encore quelques min,max et ecart type.

In [126]:
print("min")
print(fichier3pow2plus.min())
print("max")
print(fichier3pow2plus.max())
print("ecart-type")
print(fichier3pow2plus.std())

min
2001    100.002319
2002    100.000023
2003    100.007812
2004    100.004028
2005    100.006866
2006    100.008812
2007    100.001259
2008    100.007477
2009    100.009933
2010    100.000252
2011    100.027420
2012    100.001534
2013    100.002327
2014    100.013458
2015    100.000160
2016    100.274353
2017           NaN
2020    100.040000
2025    100.010000
2030    100.040000
dtype: float64
max
2001    9992270.00
2002    9982769.00
2003    9997115.00
2004    9998433.00
2005    9995796.00
2006    9999078.00
2007    9997737.00
2008    9999852.00
2009    9998954.00
2010    9999617.00
2011    9990828.00
2012    9998667.00
2013    9996921.00
2014    9999716.00
2015    9998431.00
2016    9992496.00
2017           NaN
2020    1599479.25
2025    1781898.44
2030    1967258.22
dtype: float64
ecart-type
2001    1.410469e+06
2002    1.404813e+06
2003    1.396261e+06
2004    1.408264e+06
2005    1.379149e+06
2006    1.397215e+06
2007    1.406688e+06
2008    1.419233e+06
2009    1.401934e+06
20

sauvegarde bordel